packages/check/match.js

```diff
@@ -46,39 +50,30 @@ export const check = function (value, pattern) {
  * @summary The namespace for all Match types and methods.
  */
 export const Match = {
-  Optional: function (pattern) {
-    return new Optional(pattern);
-  },
-  Maybe: function (pattern) {
-    return new Maybe(pattern);
-  },
-  OneOf: function (...args) {
-    return new OneOf(args);
-  },
+  Optional: pattern => new Optional(pattern),
+  Maybe: pattern => new Maybe(pattern),
+  OneOf: (...args) => new OneOf(args),
   Any: ['__any__'],
-  Where: function (condition) {
-    return new Where(condition);
-  },
-  ObjectIncluding: function (pattern) {
-    return new ObjectIncluding(pattern);
-  },
-  ObjectWithValues: function (pattern) {
-    return new ObjectWithValues(pattern);
-  },
+  Where: condition => new Where(condition),
```

Comment:
Unfortunately, existing code sometimes uses these properties as constructors (e.g. `new Match.Where`). Arrow functions don't have a `.prototype` and can't be called as a constructor, so I think we need to keep this code the way it was (check the blame for recent back-and-forth due to this problem). Method shorthand notation is off the table for the same reason.

Reply:
Ahh, that explains it. 👍